File: app/src/main/java/com/example/cateringplatform/adapters/FoodTypeAdapter.java
package com.example.cateringplatform.adapters;

import android.annotation.SuppressLint;
import android.graphics.Color;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.example.cateringplatform.R;

public class FoodTypeAdapter extends RecyclerView.Adapter<FoodTypeAdapter.TypeHolderFood> {
    @NonNull
    @Override
    public TypeHolderFood onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        View view = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.item_food_type,viewGroup,false);
        return new TypeHolderFood(view);
    }

    @Override
    public void onBindViewHolder(@NonNull final TypeHolderFood typeHolderFood, int i) {

        typeHolderFood.itemView.setOnClickListener(new View.OnClickListener() {
            @SuppressLint("ResourceAsColor")
            @Override
            public void onClick(View v) {
                typeHolderFood.foodTypeTV.setBackgroundResource(R.drawable.round_back_red);
                typeHolderFood.foodTypeTV.setTextColor(Color.parseColor("#FFFFFF"));
            }
        });
    }

    @Override
    public int getItemCount() {
        return 5;
    }

    public class TypeHolderFood extends RecyclerView.ViewHolder {
        private TextView foodTypeTV;
        public TypeHolderFood(@NonNull View itemView) {
            super(itemView);

            foodTypeTV = itemView.findViewById(R.id.tv_food_type);
        }
    }
}

File: app/src/main/java/com/example/cateringplatform/adapters/FoodTypeAdapter.kt
package com.example.cateringplatform.adapters

import android.content.Context
import android.graphics.Color
import android.support.v7.widget.RecyclerView
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import com.example.cateringplatform.R
import com.example.cateringplatform.models.GetCuisineModel
import kotlinx.android.synthetic.main.item_food_type.view.*

class FoodTypeAdapter(val context: Context?,val getCuisineModel: GetCuisineModel?) : RecyclerView.Adapter<FoodTypeAdapter.TypeHolderFood>() {

    private var row_index:Int=-1

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): TypeHolderFood {
        val view = LayoutInflater.from(viewGroup.context).inflate(R.layout.item_food_type, viewGroup, false)
        return TypeHolderFood(view)
    }

    override fun onBindViewHolder(typeHolderFood: TypeHolderFood, i: Int) {

            typeHolderFood.itemView.tv_food_type.text=getCuisineModel?.response?.cuisine_list?.get(i)?.cuisine_name

        if (row_index==i){
            typeHolderFood.foodTypeTV.setBackgroundResource(R.drawable.round_back_red)
            typeHolderFood.foodTypeTV.setTextColor(Color.parseColor("#FFFFFF"))
        } else{
            typeHolderFood.foodTypeTV.setBackgroundResource(R.drawable.roun_back_default)
            typeHolderFood.foodTypeTV.setTextColor(Color.parseColor("#000000"))
        }

        typeHolderFood.itemView.setOnClickListener(View.OnClickListener {

            row_index = i
            notifyDataSetChanged()
    })
            Log.i("555",""+getCuisineModel?.response?.cuisine_list?.get(i)?.cuisine_name)
    }

    override fun getItemCount(): Int {
        return getCuisineModel?.response?.cuisine_list!!.size
    }

    inner class TypeHolderFood(itemView: View) : RecyclerView.ViewHolder(itemView) {
         val foodTypeTV: TextView


        init {

            foodTypeTV = itemView.findViewById(R.id.tv_food_type)
        }
    }
}
